app/scripts/views/sign_up.js

```diff
@@ -269,7 +319,7 @@ define(function (require, exports, module) {
       this.navigate('cannot_create_account');
     },
 
-    _initAccount: function () {
+    _initAccount: function (action) {
```

Comment:
Instead of passing in the `action` name, maybe just pass in the continuation function.